qcodes/config/config.py

```diff
@@ -156,7 +156,9 @@ def update_config(self, path: str=None) -> dict:
             schema_file = os.path.join(self.config_file_path,
                                        self.schema_file_name)
             self._update_config_from_file(config_file, schema_file, config)
-
+        if config is None:
```

Comment:
Yes but that is only because the default config file always exists. There is no way that the type system can know that. A sound type system will have to assume that loading a file from disc can always fail. You could just replace this with a cast that throws away the Optional of the type. I think either way is equally good

Reply:
Edit: This is more because the class sets the config member variable to None and later overwrites it. The type system has trouble seeing that this code is only hit after the config variable has been set. I don't think there is a good solution for this other than rewriting the whole config class to remove this antipattern

Reply:
Oh, I see, thanks. Well, we'll do it at some point in another PR :)